### DISCUSSION TITLE
Dioxus Desktop Performance

### DISCUSSION THREAD

**Discussion by John0x** [DioxusLabs/dioxus#810] [General]
Hey, I'm currently looking for a GUI framework for a new project and Dioxus looks pretty interesting. I'm kinda worried about the performance though.
From my understanding the desktop version is using tauri as the renderer, which would mean that everything has to be sent to the webview to be rendered.
Is this viable for applications that have to display a lot of big files, like an image editor/viewer?
Let's say we load an image using rust, do some processing and need to display that processed image without writing it to disk. I could probably display it using the image tag with the data as base64, but that would need to be serialized and sent to the webview via some rpc technology and therefore wouldn't be very performant, right?

**Comment by John0x** [comment]
@jkelleyrtp do you happen to have some insight to this? :)

**Comment by ealmloff** [comment]
We currently use json for serialization between Dioxus and the webview. We could use a binary format which would improve performance (see [this issue](https://github.com/DioxusLabs/dioxus/issues/695)). Once this change is implemented performance should be similar or better than Dioxus web.

Until then you have two options:

1) You could package your Dioxus-web application into a webview with the code running as WASM. There is still a serialization boundary between Dioxus <-> js, but compared to other web js frameworks, Dioxus performs well
2) You could use a [custom request protocal](https://docs.rs/dioxus-desktop/latest/dioxus_desktop/struct.Config.html#method.with_custom_protocol) to stream the image as binary data into the webview and use a html img element to renderer it which should be very performant, but depending on how you plan to render the image it might not be applicable.

In the future, this will be an area where Blitz will be very useful because the rust code can directly access the renderer

**Comment by jon-ton** [comment]
Excited for this. Being able to render directly from a format like Apache Arrow would be sweet for data visualization applications.
